crates/bevy_ecs/src/schedule/executor/mod.rs

```diff
@@ -119,3 +119,51 @@ pub(super) fn is_apply_deferred(system: &BoxedSystem) -> bool {
     // deref to use `System::type_id` instead of `Any::type_id`
     system.as_ref().type_id() == apply_deferred.system_type_id()
 }
+
+/// These functions hide the bottom of the callstack from `RUST_BACKTRACE=1` (assuming the default panic handler is used).
+/// The full callstack will still be visible with `RUST_BACKTRACE=full`.
+/// They are specialized for `System::run` & co instead of being generic over closures because this avoids an
+/// extra frame in the backtrace.
```

Comment:
This should probably mention that this is reliant on `std::sys_common::__rust_begin_short_backtrace` not changing, which is unstable.

Reply:
```suggestion
/// extra frame in the backtrace.
///
/// This is reliant on unstable behavior in Rust's default panic handler and the name of 
/// `std::sys_common::__rust_begin_short_backtrace`, and may break if the function is renamed.
```